From 31545c23277cd54a1edd41c85d8255fb589158e3 Mon Sep 17 00:00:00 2001
From: Joseph Myers <joseph@codesourcery.com>
Date: Mon, 29 Jun 2015 14:38:46 +0000
Subject: [PATCH] Update headers for Linux 4.0, 4.1 definitions.

This patch updates installed glibc headers for new definitions from
Linux 4.0 and 4.1 that seem relevant to glibc headers.  In addition, I
noticed that PF_IB / AF_IB, added in Linux 3.11, were missing for no
obvious reason, so added those as well.

Tested for x86_64 (testsuite, and that installed stripped shared
libraries are unchanged by the patch).

	* sysdeps/unix/sysv/linux/bits/in.h (IP_CHECKSUM): New macro.
	* sysdeps/unix/sysv/linux/bits/socket.h (PF_IB): Likewise.
	(PF_MPLS): Likewise.
	(AF_IB): Likewise.
	(AF_MPLS): Likewise.
	* sysdeps/unix/sysv/linux/sys/mount.h (MS_LAZYTIME): New enum
	value and macro.
	(MS_RMT_MASK): Include MS_LAZYTIME.
---
 sysdeps/unix/sysv/linux/bits/in.h     |  1 +
 sysdeps/unix/sysv/linux/bits/socket.h |  4 ++++
 sysdeps/unix/sysv/linux/sys/mount.h   |  5 ++++-
 4 files changed, 20 insertions(+), 1 deletion(-)

diff --git a/sysdeps/unix/sysv/linux/bits/in.h b/sysdeps/unix/sysv/linux/bits/in.h
index 43db11c..f684736 100644
--- a/sysdeps/unix/sysv/linux/bits/in.h
+++ b/sysdeps/unix/sysv/linux/bits/in.h
@@ -91,6 +91,7 @@
 
 #define IP_MINTTL       21
 #define IP_NODEFRAG     22
+#define IP_CHECKSUM     23
 
 /* IP_MTU_DISCOVER arguments.  */
 #define IP_PMTUDISC_DONT   0	/* Never send DF frames.  */
diff --git a/sysdeps/unix/sysv/linux/bits/socket.h b/sysdeps/unix/sysv/linux/bits/socket.h
index 88b4cfc..5f70488 100644
--- a/sysdeps/unix/sysv/linux/bits/socket.h
+++ b/sysdeps/unix/sysv/linux/bits/socket.h
@@ -68,6 +68,8 @@ typedef __socklen_t socklen_t;
 #define PF_PPPOX	24	/* PPPoX sockets.  */
 #define PF_WANPIPE	25	/* Wanpipe API sockets.  */
 #define PF_LLC		26	/* Linux LLC.  */
+#define PF_IB		27	/* Native InfiniBand address.  */
+#define PF_MPLS		28	/* MPLS.  */
 #define PF_CAN		29	/* Controller Area Network.  */
 #define PF_TIPC		30	/* TIPC sockets.  */
 #define PF_BLUETOOTH	31	/* Bluetooth sockets.  */
@@ -113,6 +115,8 @@ typedef __socklen_t socklen_t;
 #define AF_PPPOX	PF_PPPOX
 #define AF_WANPIPE	PF_WANPIPE
 #define AF_LLC		PF_LLC
+#define AF_IB		PF_IB
+#define AF_MPLS		PF_MPLS
 #define AF_CAN		PF_CAN
 #define AF_TIPC		PF_TIPC
 #define AF_BLUETOOTH	PF_BLUETOOTH
diff --git a/sysdeps/unix/sysv/linux/sys/mount.h b/sysdeps/unix/sysv/linux/sys/mount.h
index 96da656..00b7135 100644
--- a/sysdeps/unix/sysv/linux/sys/mount.h
+++ b/sysdeps/unix/sysv/linux/sys/mount.h
@@ -78,6 +78,8 @@ enum
 #define MS_I_VERSION	MS_I_VERSION
   MS_STRICTATIME = 1 << 24,	/* Always perform atime updates.  */
 #define MS_STRICTATIME	MS_STRICTATIME
+  MS_LAZYTIME = 1 << 25,	/* Update the on-disk [acm]times lazily.  */
+#define MS_LAZYTIME	MS_LAZYTIME
   MS_ACTIVE = 1 << 30,
 #define MS_ACTIVE	MS_ACTIVE
   MS_NOUSER = 1 << 31
@@ -85,7 +87,8 @@ enum
 };
 
 /* Flags that can be altered by MS_REMOUNT  */
-#define MS_RMT_MASK (MS_RDONLY|MS_SYNCHRONOUS|MS_MANDLOCK|MS_I_VERSION)
+#define MS_RMT_MASK (MS_RDONLY|MS_SYNCHRONOUS|MS_MANDLOCK|MS_I_VERSION \
+		     |MS_LAZYTIME)
 
 
 /* Magic mount flag number. Has to be or-ed to the flag values.  */
-- 
2.4.5

